pkg/operator/apis/preview.aro.openshift.io/v1alpha1/previewfeature_types.go

```diff
@@ -24,13 +24,23 @@ type NSGFlowLogs struct {
 	// +kubebuilder:validation:Enum=1;2
 	Version int `json:"version,omitempty"`
 
-	// StorageAccountResourceID should be defined if BYO-Storage account is used.
+	// FlowLogName defines the name of the flow log.
+	FlowLogName string `json:"flowLogName,omitempty"`
+
+	// NetworkSecurityGroupResourceID defines the NetworkSecurityGroup for which the flow logs
+	// should be collected. It must be in the same resource group as the cluster.
+	NetworkSecurityGroupResourceID string `json:"networkSecurityGroupResourceID,omitempty"`
+
+	// Location specifies the location of the network watcher (TODO alternatively require network watcher ID).
+	Location string `json:"location,omitempty"`
```

Comment:
I think it is better to require network watcher id as will not always be `NetworkWatcher_%s` RG: customers can create custom ones and can remove default ones.